### PR TITLE
add support for kubernetes authentication backend following => https:…

### PR DESCRIPTION
**Pull request by visheyra**
This implement in HVAC library the kubernetes auth backend API described here => https://www.vaultproject.io/api/auth/kubernetes/index.html .

### CODE REVIEW

File: hvac/v1/__init__.py [hunk 1]
```diff
@@ -978,6 +977,80 @@ def auth_approle(self, role_id, secret_id=None, mount_point='approle', use_token
...
+
+    def get_kubernetes_configuration(self, mount_point='kubernetes'):
+        """
+        GET /auth/kubernetes/login
+        """
+
+        url = '/v1/auth/{0}/config'.format(mount_point)
+        return self._get(url).json()
+
+    def auth_kubernetes(self, host, role, jwt, mount_point='kubernetes'):
```
Comment (smatochkin) [inline hvac/v1/__init__.py]: Thanks for adding Kubernetes auth!

`host` parameter in `auth_kubernetes` is not used. What is it intended for?

**Comment by visheyra** [comment]
My bad, must been a copy paste issue. I've fixed it on my master branch

**Comment by visheyra** [comment]
up @smatochkin ?

**Comment by visheyra** [comment]
Hello :)

It's been a while since this PR is open. Do you need me to do anything on my branch ?

**Comment by jeffwecan** [comment]
@visheyra Apologies for the delays here. If you're willing to add some tests for these new methods I'd be happy to get them merged in. If I can provide any guidance along those lines, just let me know!

**Comment by visheyra** [comment]
Sorry I've seen the comment just today ... I'll will look at the tests this week. I'll keep in touch with you if I need any guidance.

Thanks for not forgiving me ;)

**Comment by jeffwecan** [comment]
Hey @visheyra, there might be some other closed PRs we would like to see released sometime this week. I would like these new methods to be included in that release as well. If you have some staged work around adding tests, please let me know. Otherwise, if it is okay with you, I can attempt to add in those tests and push them up to your fork's master branch so they get included here.

**Comment by jeffwecan** [comment]
@visheyra: Since I didn't hear back from you for a bit, I went ahead and pulled down your fork and opened up a new PR with the tests and whatnot added:  https://github.com/ianunruh/hvac/pull/210

**Pull request closed without merging**